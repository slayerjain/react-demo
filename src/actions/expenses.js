
// TODO Remove
const addExpense = expense => ({
  type: 'ADD_EXPENSE',
  expense,
});

const startAddExpense = ({
  description = '',
  note = '',
  amount = 0,
  createdAt = 0,
} = {}) => ({
  type: 'START_ADD_EXPENSE',
  expense: {
    description,
    note,
    amount,
    createdAt,
  },
});

// const removeExpense = ({ id } = {}) => ({
//   type: 'REMOVE_EXPENSE',
//   id,
// });

const startRemoveExpense = ({ id } = {}) => ({
  type: 'START_REMOVE_EXPENSE',
  id,
});

const startEditExpense = (id, updates) => ({
  type: 'START_EDIT_EXPENSE',
  id,
  updates,
});

const startSetExpenses = () => ({
  type: 'START_SET_EXPENSES',
});

export { addExpense, startEditExpense, startAddExpense, startSetExpenses, startRemoveExpense };
