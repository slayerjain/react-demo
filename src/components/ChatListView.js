import React from 'react';
import { ChatList } from 'react-chat-elements';


const ChatListView = () => (
  <div className="content-container--chat-list">
    <ChatList
      className="chat-list content-container--chat-list-item"
      dataSource={[
        {
          avatar: 'https://facebook.github.io/react/logo-og.png',
          alt: 'Reactjs',
          title: 'Shubham',
          subtitle: 'What are you doing?',
          date: new Date(),
          unread: 0,
        },
        {
          alt: 'Reactjs',
          title: 'Pankaj',
          subtitle: 'Here are my Powers',
          date: new Date(),
          unread: 2,
        },
        ]}
    />
  </div>
);

export default ChatListView;
