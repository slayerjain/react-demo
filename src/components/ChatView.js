import React from 'react';
import { MessageList, Input, Button } from 'react-chat-elements';
const ChatView = () => (
  <div className="content-container--message-list">
    <MessageList
      className="message-list"
      lockable
      toBottomHeight="100%"
      dataSource={[
        {
          position: 'right',
          type: 'text',
          text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
          date: new Date(),
        },
        {
          position: 'left',
          type: 'text',
          text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
          date: new Date(),
        },
        ]}
    />
    <Input
      className="content-container--message-input"
      placeholder="Type here..."
      autoHeight
      multiline
      rightButtons={
        <Button
          color="white"
          text="Send"
        />
      }
    />
  </div>
);

export default ChatView;
