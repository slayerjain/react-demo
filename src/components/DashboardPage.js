import React from 'react';
import { SideBar } from 'react-chat-elements';

const DashboardPage = () => (
  <SideBar
    top={
      <div>'TOP' area</div>
    }
    center={
      <div>'CENTER' area</div>
    }
    bottom={
      <div>'BOTTOM' area</div>
    }
  />
);

export default DashboardPage;
