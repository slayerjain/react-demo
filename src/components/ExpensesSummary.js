import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';

import selectExpenses from '../selectors/expenses';
import totalExpenses from '../selectors/total-expenses';

export const ExpensesSummary = ({ expenseCount, expenseTotal }) => (
  <div>
    {
      `Viewing ${expenseCount} expense${expenseCount > 1 ? 's' : ''} totaling ${numeral(expenseTotal/ 100).format('$0,0.00')}`
    }
  </div>
);

const mapStateToProps = state => {
  const selectedExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: selectedExpenses.length,
    expenseTotal: totalExpenses(selectedExpenses),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
