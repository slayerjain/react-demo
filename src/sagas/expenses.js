import { call, put, takeEvery, all } from 'redux-saga/effects';
import { AddToDBException } from '../errors/errors';
import { addExpenseToFirebase, getExpensesFromFirebase, removeExpensesFromFirebase, editExpenseFromFirebase } from '../async/firebase';


// worker Saga: will be fired on START_ADD_EXPENSE actions
export function* addExpense(action = { expense: {} }) {
  const {
    description = '',
    amount = 0,
    createdAt = 0,
    note = '',
  } = action.expense;
  const expense = {
    description, amount, createdAt, note,
  };
  const id = yield call(addExpenseToFirebase, expense);
  if (!id) {
    throw AddToDBException('Error getting new id of added document');
  }
  yield put({
    type: 'ADD_EXPENSE',
    expense: {
      id,
      ...expense,
    },
  });
}

export function* setExpense() {
  const expenses = yield call(getExpensesFromFirebase);
  if (expenses) {
    yield put({
      type: 'SET_EXPENSES',
      expenses,
    });
  }
}

export function* removeExpense({ id }) {
  const idDeleted = yield call(removeExpensesFromFirebase, id);
  if (idDeleted) {
    yield put({
      type: 'REMOVE_EXPENSE',
      id,
    });
  }
}

export function* editExpense({ id, updates }) {
  const isUpdated = yield call(editExpenseFromFirebase, id, updates);
  if (isUpdated) {
    yield put({
      type: 'EDIT_EXPENSE',
      id,
      updates,
    });
  }
}

/*
  Starts fetchUser on each dispatched `USER_FETCH_REQUESTED` action.
  Allows concurrent fetches of user.
*/
function* mySaga() {
  yield all([
    takeEvery('START_ADD_EXPENSE', addExpense),
    takeEvery('START_SET_EXPENSES', setExpense),
    takeEvery('START_REMOVE_EXPENSE', removeExpense),
    takeEvery('START_EDIT_EXPENSE', editExpense),
  ]);
}

export default mySaga;
