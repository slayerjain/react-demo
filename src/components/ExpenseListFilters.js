import React from 'react';
import uuid from 'uuid'
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';


import { setEndDate, setStartDate, setTextFilter, sortByAmount, sortByDate } from '../actions/filters'

export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null,
  }

  onDatesChanged = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };

  onFocusChanged = calendarFocused => {
    this.setState(() => ({ calendarFocused }));
  };

  onTextChange = e => {
    this.props.setTextFilter(e.target.value);
  }

  onSortChange = e => {
    if (e.target.value === 'date') {
      this.props.sortByDate();
    } else if (e.target.value === 'amount') {
      this.props.sortByAmount();
    }
  }

  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={this.onTextChange}
        />
        <select
          value={this.props.filters.sortBy}
          onChange={this.onSortChange}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate}
          startDateId={uuid()}
          endDate={this.props.filters.endDate}
          endDateId={uuid()}
          onDatesChange={this.onDatesChanged}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChanged}
          showClearDates={true}
          numberOfMonths={1}
          isOutsideRange={() => (false)}
        />
      </div>
    )
  }
}

const mapStateToProps = state => ({
  filters: state.filters,
});

const mapDispatchToProps = dispatch => ({
  setTextFilter: text => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount()),
  setStartDate: startDate => dispatch(setStartDate(startDate)),
  setEndDAte: endDate => dispatch(setEndDate(endDate)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
