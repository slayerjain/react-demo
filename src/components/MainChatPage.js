import React from 'react';
import ChatListView from './ChatListView';
import ChatView from './ChatView';

const MainChatPage = () => (
  <div className="content-container content-container--chat-view-split">
    <ChatListView />
    <ChatView />
  </div>
);

export default MainChatPage;
