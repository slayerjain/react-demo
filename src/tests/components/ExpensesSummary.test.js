import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';
import totalExpenses from '../../selectors/total-expenses';

test('should render ExpensesSummary with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary
    expenseCount={[expenses[0]].length}
    expenseTotal={totalExpenses([expenses[0]])}
  />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with expenses', () => {
  const wrapper = shallow(<ExpensesSummary
    expenseCount={expenses.length}
    expenseTotal={totalExpenses(expenses)}
  />);
  expect(wrapper).toMatchSnapshot();
});
