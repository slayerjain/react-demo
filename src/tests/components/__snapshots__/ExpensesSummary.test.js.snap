// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expense 1`] = `
<div>
  Viewing 1 expense totaling $1.23
</div>
`;

exports[`should render ExpensesSummary with expenses 1`] = `
<div>
  Viewing 3 expenses totaling $4,901.23
</div>
`;
